homeassistant/components/roku/media_player.py

```diff
@@ -238,16 +238,33 @@ async def search(self, keyword):
         """Emulate opening the search screen and entering the search keyword."""
         await self.coordinator.roku.search(keyword)
 
+    async def async_get_browse_image(
+        self, media_content_type, media_content_id, media_image_id=None
+    ):
+        """Fetch media browser image to serve via proxy."""
+        if media_content_type == MEDIA_TYPE_APP and media_content_id:
+            image_url = self.coordinator.roku.app_icon_url(media_content_id)
+            result = await self._async_fetch_image(image_url)
+            if result == (None, None):
+                _LOGGER.debug("Error retrieving proxied image from %s", image_url)
+            return result
+
+        return (None, None)
+
     async def async_browse_media(self, media_content_type=None, media_content_id=None):
         """Implement the websocket media browsing helper."""
+
+        def _get_thumbnail_url(*args, **kwargs):
+            return self.get_browse_image_url(*args, **kwargs)
```

Comment:
Should we add same logic as #43089

Reply:
so the issue with roku images is they arent accessible over https so wed need a bit deeper of a check to check if internal and non-https request.

Reply:
Oh right. Squeezebox will have the same issue.

Reply:
Actually, that is not a problem. Locally there can't be valid SSL anyway

Reply:
ah ok. I can look into doing an update tonight for internal check